apps/dashboard/src/components/workflow-editor/step-utils.ts

```diff
@@ -58,10 +67,25 @@ export const updateStepInWorkflow = (
   };
 };
 
-export const createStep = (type: StepTypeEnum): Step => ({
-  name: STEP_TYPE_LABELS[type] + ' Step',
-  stepId: '',
-  slug: '_st_',
-  type,
-  _id: crypto.randomUUID(),
-});
+export const createStep = (type: StepTypeEnum): StepCreateDto => {
+  const controlValue: Record<string, unknown> = {};
+
+  if (type === StepTypeEnum.DIGEST) {
+    controlValue.amount = DEFAULT_CONTROL_DIGEST_AMOUNT;
+    controlValue.unit = DEFAULT_CONTROL_DIGEST_UNIT;
+    controlValue.digestKey = DEFAULT_CONTROL_DIGEST_DIGEST_KEY;
+    controlValue.cron = DEFAULT_CONTROL_DIGEST_CRON;
+  }
+
+  if (type === StepTypeEnum.DELAY) {
+    controlValue.amount = DEFAULT_CONTROL_DELAY_AMOUNT;
+    controlValue.unit = DEFAULT_CONTROL_DELAY_UNIT;
+    controlValue.type = DEFAULT_CONTROL_DELAY_TYPE;
+  }
+
+  return {
+    name: STEP_TYPE_LABELS[type] + ' Step',
+    type,
+    controlValues: controlValue,
+  };
+};
```

Comment:
Why This Change Is Needed and What Is Happening in the Current Implementation  

The main reason for this change is that we cannot create a step with control values during step creation. This limitation causes unexpected behavior, like the [bug](https://linear.app/novu/issue/NV-5124/dashboard-save-control-value-on-add-step) we are experiencing.  

Currently, we cannot move schemas with default values to the `Shared` package and reuse them in the `Dashboard` because these schemas depend on `Zod` and `Zod-to-JSON-Schema`.  

The solution we originally wanted—to move default values from the `UiSchema` to the `dataSchema`—is not supported natively. The solution should be `Zod` schema with default values (used by the `Dashboard`) and make certain attributes, like `body`, required for generating `issues`. However, this approach fails because `zod-to-json-schema` removes the default values meaning we will need to manipulate the schema manually.  

An improved solution would reduce the amount of code. If we store the default values during step creation, we wouldn't need to handle parsing them (placeholders/defaults) in the `Dashboard` or maintain "placeholders" or default values in the API.

**Next Steps**  
- Remove placeholders from the API.  
- Eliminate placeholder parsing in the `Dashboard`.